plugins/kubernetes-backend/src/service/getKubernetesObjectsForServiceHandler.ts

```diff
@@ -92,7 +79,10 @@ export const handleGetKubernetesObjectsForService: GetKubernetesObjectsForServic
       .join(', ')}]`,
   );
 
-  const labelSelector = parseLabelSelector(requestBody.entity);
+  const labelSelector: string =
+    requestBody.entity?.metadata?.annotations?.[
+      'backstage.io/kubernetes-label-selector'
+    ] || `backstage.io/kubernetes-id=${requestBody.entity.metadata.name}`;
```

Comment:
Just noting that the line above is very defensive with `entity?.metadata?.` but this one just gets it plain. Gonna assume that the one on this line is ok.